### PR TITLE
dbp-608-Fixed commit SHA for external Jobs

### PR DESCRIPTION
**Pull request by sahassou**
For external jobs, it is recommended to use fixed commit SHA references.

TODO:

Identify all external jobs currently referenced in our CI/CD workflows.
Update the workflow configurations to reference the external jobs using their respective commit SHAs instead of semantic versions.
AC:

All references to external jobs in the CI/CD workflows must use a fixed commit SHA.
 

# Description

## Approval for review

- [ ] All points were discussed with the ticket creator, support-team or product owner. The code upholds all quality guidelines from the PR-template.

> Notice: Please remove the WIP label if the PR is ready to review, otherwise nobody will review it.

### CODE REVIEW

File: .github/workflows/delete-namespace.yml [hunk 1]
```diff
@@ -63,12 +63,12 @@ jobs:
     runs-on: ubuntu-latest
     steps:
     - name: checkout repo
-      uses: actions/checkout@v2
+      uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11  #v4.1.1
```
Comment (M-Schiborr) [inline .github/workflows/delete-namespace.yml:66]: Have you tested everywhere that this newer major version works for us like before?

File: .github/workflows/delete-namespace.yml [hunk 2]
```diff
@@ -63,12 +63,12 @@ jobs:
...
       with:
         repository: 'dBildungsplattform/spsh-app-deploy'
 
     - name: Set up kubectl and Helm
-      uses: azure/setup-kubectl@v3
+      uses: azure/setup-kubectl@3e0aec4d80787158d308d7b364cb1b702e7feb7f #v4.0.0
```
Comment (M-Schiborr) [inline .github/workflows/delete-namespace.yml:71]: Same for this action: Have you tested everywhere that this newer major version works for us like before?

File: .github/workflows/delete-namespace.yml (hunk 1 above)
Comment (sahassou) [reply .github/workflows/delete-namespace.yml:66]: I tested the new commit hash one the deployment of  the spsh-client Helm chart. The CI/CD pipeline ran successfully. You can check this workflow. By the way, trivy scan failed, but the cause is not the commit hash; rather, it's due to CVE-2023-52425.
https://github.com/dBildungsplattform/schulportal-client/actions/runs/7895165196

File: .github/workflows/delete-namespace.yml (hunk 2 above)
Comment (sahassou) [reply .github/workflows/delete-namespace.yml:71]: Same response here

File: .github/workflows/delete-namespace.yml (hunk 1 above)
Comment (aimee-889) [reply .github/workflows/delete-namespace.yml:66]: this has not been tested here: https://github.com/dBildungsplattform/spsh-app-deploy/actions/workflows/delete-namespace-manual.yml

Comment (sahassou) [reply .github/workflows/delete-namespace.yml:66]: I have tested it on the 'delete-manual-namespace
https://github.com/dBildungsplattform/spsh-app-deploy/actions/runs/7914749618

**Review by aimee-889 — APPROVED** [review]

**Pull request merged**